core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -147,9 +145,12 @@ public void spill() throws IOException {
   }
 
   private long getMemoryUsage() {
-    return sorter.getMemoryUsage() + (allocatedPages.size() * (long) PAGE_SIZE);
+    long totalPageSize = 0;
+    for (MemoryBlock page : allocatedPages) {
```

Comment:
This is only called when spilling and the corresponding methods in the other classes are only called in debugging code which isn't called in normal operation.  Therefore, I'm not planning to change this in this patch.